app/middleware/renderer.js

```diff
@@ -15,7 +18,36 @@ function age(req, res) {
 }
 
 function choose(req, res) {
-  res.render('choose');
+  return res.render('choose');
+}
+
+function location(req, res) {
+  if (!res.locals.correctLocationParams) {
+    return res.render('error');
+  }
+  return res.render('location');
+}
+
+function results(req, res) {
+  res.render('results');
+}
+
+function emptyPostcode(req, res) {
+  log.debug('Empty search');
+  res.locals.errorMessage = messages.emptyPostcodeMessage();
+  location(req, res);
+}
+
+function invalidPostcode(req, res, loc) {
+  log.debug({ loc }, 'Location failed validation');
```

Comment:
Can the debug payload be changed to `{ location: loc }` so that there is consistency in field for which the `location` search term is recorded against please.